### PR TITLE
Update AKS Guidance

### PR DESCRIPTION
**Pull request by ibersanoMS**
# Overview/Summary

Updated guidance for AKS to include [these items](https://microsoft-my.sharepoint.com/:x:/r/personal/maprado_microsoft_com/Documents/Azure%20Core/Resiliency/Documentation/Action%20Plan%20-%20Services%20Hub%20Backlog%20items%20-%20AKS%20-%20Template.xlsx?d=wb0c5024f548e4c29b61fddbf5b6852ab&csf=1&web=1&e=5TcTox)

## This PR fixes/adds/changes/removes

1. Adds AKS guidance items 8-31

### Breaking Changes

None

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by ibersanoMS** [comment]
Do yall need a screenshot for each ARG query I added? I tested them when I wrote up the guidance, but I can include screenshots of the output if needed.

**Comment by ejhenry** [comment]
> Do yall need a screenshot for each ARG query I added? I tested them when I wrote up the guidance, but I can include screenshots of the output if needed.

Hi @ibersanoMS, thanks for the PR! Yes, please include a screenshot of successful testing of each query that has been added. Also, please include a brief comment in each ARG query explaining what it is doing.

**Comment by ibersanoMS** [comment]
Screenshots for AKS queries 8,11 ,13,15, 20, 22, 23, 24, 27
[Azure PRR - AKS_8-31_Query Screenshots.zip](https://github.com/Azure/Azure-Proactive-Resiliency-Library/files/13877478/Azure.PRR.-.AKS_8-31_Query.Screenshots.zip)

File: docs/content/services/container/aks/_index.md [hunk 1]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+
+### AKS-8 - Control traffic flow with network policies
+
+**Category: Access & Security**
+
+**Impact: Low**
+
+**Guidance**
+
+Use network policies to allow or deny traffic to pods. By default, all traffic is allowed between pods within a cluster. For improved security, define rules that limit pod communication. Network policies allow or deny traffic to the pod based on settings such as assigned labels, namespace, or traffic port. Network policies are a cloud-native way to control the flow of traffic for pods. As pods are dynamically created in an AKS cluster, required network policies can be automatically applied. To use network policy, enable the feature when you create a new AKS cluster. You can't enable network policy on an existing AKS cluster. Plan ahead to enable network policy on the necessary clusters.
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:248]: Could you please clarify how this impacts reliability, it looks to be more aligned to WAF security pillar.

File: docs/content/services/container/aks/_index.md [hunk 2]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+
+### AKS-10 - Remediate AKS non-compliant Azure Policies
+
+**Category: Compliance**
+
+**Impact: Low**
+
+**Guidance**
+
+It’s important to keep your AKS cluster secure for the applications that you run. Azure Policy makes it possible to manage and report on the compliance state of your Kubernetes clusters from one place, enforces organizational standards, built-in security policies and assess compliance at-scale.
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:301]: Could you please adjust the wording here as well to incorporate impact to reliability.

File: docs/content/services/container/aks/_index.md [hunk 3]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+### AKS-15 - Deploy a private AKS cluster if public access is not required
+
+
+**Category: Security**
+
+**Impact: Low**
+
+**Guidance**
+
+As you manage clusters in Azure Kubernetes Service (AKS), workload and data security is a key consideration. When you run multi-tenant clusters using logical isolation, you especially need to secure resource and workload access. Minimize the risk of attack by isolating the AKS cluster from public access.
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:437]: This one probably is specific to security and not reliability, unless you disagree?

File: docs/content/services/container/aks/_index.md [hunk 4]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+
+{{< collapse title="Show/Hide Query/Script" >}}
+
+{{< code lang="sql" file="code/aks-15/aks-15.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
+
+### AKS-16 - Keep a consistent configuration between regions with the API Server Authorized IP Addresses
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:456]: Same with this one as well?

File: docs/content/services/container/aks/_index.md [hunk 5]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+
+{{< collapse title="Show/Hide Query/Script" >}}
+
+{{< code lang="sql" file="code/aks-22/aks-22.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
+
+### AKS-23 - Configure the API server Public IP access for public clusters
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:637]: This appears to be focused directly towards security and not reliability

File: docs/content/services/container/aks/_index.md [hunk 6]
```diff
@@ -212,3 +236,623 @@ AKS is increasingly being used for stateful applications that require a backup s
...
+
+{{< collapse title="Show/Hide Query/Script" >}}
+
+{{< code lang="sql" file="code/aks-27/aks-27.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
+
+### AKS-28 - Use the Azure KeyVault provider for Secrets Store CSI Driver
```
Comment (oZakari) [inline docs/content/services/container/aks/_index.md:760]: Focused specifically towards security, please adjust wording to how it impacts reliability.

**Review by oZakari — CHANGES_REQUESTED** [review]
Hi @ibersanoMS, great work on this PR and thank you! Just noted some recommendations that appear to be explicitly oriented towards security. Could you please go through these and either update the guidance to reflect how it impacts reliability or potentially pull them out if necessary?

**Comment by ibersanoMS** [comment]
@oZakari Thanks for going through my PR! Yes, I can go through make those edits. The reason the id doesn't display is because if you put the id first in the order of properties that get displayed, it hides the id when you turn on Formatted results. If you turn off Formatted results, it will show the id. For the screenshots I had that on to hide my subscription id.

File: docs/content/services/container/aks/_index.md (hunk 1 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:248]: Removed

File: docs/content/services/container/aks/_index.md (hunk 2 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:301]: Adjusted wording

File: docs/content/services/container/aks/_index.md (hunk 3 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:437]: Removed

File: docs/content/services/container/aks/_index.md (hunk 4 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:456]: Removed

File: docs/content/services/container/aks/_index.md (hunk 5 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:637]: Removed

File: docs/content/services/container/aks/_index.md (hunk 6 above)
Comment (ibersanoMS) [reply docs/content/services/container/aks/_index.md:760]: Removed

**Comment by ibersanoMS** [comment]
New screenshots for now AKS 10, 12, 17, 19, 20 and 23
[Azure PRR - AKS_10_23_Query Screenshots.zip](https://github.com/Azure/Azure-Proactive-Resiliency-Library/files/13905336/Azure.PRR.-.AKS_10_23_Query.Screenshots.zip)

**Review by oZakari — APPROVED** [review]

**Comment by oZakari** [comment]
Thanks much @ibersanoMS! Also, thanks for correcting me about the screenshots, didn't notice the formatting toggle :D

I have approved the PR.

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution @ibersanoMS! It looks as though several of the ARG queries are returning results both of resources that do align and resources that do not align to the recommendation. The queries should be written to only return resources that do not align to the recommendation. For example, we would want the AKS-12 query to only return clusters with local accounts enabled.

**Comment by ibersanoMS** [comment]
Thanks @ejhenry! I updated the queries to only return the non-compliant clusters. Apologies - I missed that in the docs. I've attached new screenshots here.
[Azure PRR - AKS_10_23_Revised_Query Screenshots.zip](https://github.com/Azure/Azure-Proactive-Resiliency-Library/files/13907789/Azure.PRR.-.AKS_10_23_Revised_Query.Screenshots.zip)

**Comment by oZakari** [comment]
Thanks for the call out @ibersanoMS, just created a separate PR as we agree that the query requirement was not clear in that regard: https://github.com/Azure/Azure-Proactive-Resiliency-Library/pull/208

**Review by oZakari — APPROVED** [review]

**Review by ejhenry — APPROVED** [review]
LGTM 👍 Thanks @ibersanoMS !

**Pull request merged**